### PR TITLE
finished adding react-query for server storage handling

### PR DESCRIPTION
**Pull request by Abdi-MS**

### CODE REVIEW

File: src/components/NewToDo/NewToDo.css [hunk 1]
```diff
@@ -1,3 +1,7 @@
+/* * {
```
Comment (zain-sattar) [inline src/components/NewToDo/NewToDo.css]: Remove these unused comments

File: src/components/NewToDo/NewToDo.jsx [hunk 2]
```diff
@@ -2,9 +2,27 @@ import React, { useState } from "react";
 import { TextField } from "@mui/material";
 import "./NewToDo.css";
 import { v4 as uuidv4 } from "uuid";
+import { addToDo as reduxAddToDo } from "../../store/store";
```
Comment (zain-sattar) [inline src/components/NewToDo/NewToDo.jsx]: why are we still using redux?

File: src/components/OneToDo/OneToDo.jsx [hunk 3]
```diff
@@ -2,30 +2,77 @@ import { Checkbox, IconButton, TextField, Typography } from "@mui/material";
...
 
-function OneToDo({ index, toDoList, editToDo, deleteToDo }) {
   const [editingToDo, setEditingToDo] = useState(false);
   const [toDoStatus, setToDoStatus] = useState(false);
 
   const editField = useRef();
 
+  const toDoList = useToDoList().toDoList;
+
+  useEffect(() => {
```
Comment (zain-sattar) [inline src/components/OneToDo/OneToDo.jsx]: why are you using this state and the useEffect?
i mean we can directly use the todo.checked property

File: src/components/OneToDo/OneToDo.jsx [hunk 4]
```diff
@@ -2,30 +2,77 @@ import { Checkbox, IconButton, TextField, Typography } from "@mui/material";
...
+      editToDo(editedToDo);
+      editMutation.mutate(editedToDo);
+    }
+  };
+
   const editToDoHandler = () => {
-    editToDo({
+    const editedToDo = {
       text: editField.current.value,
+      checked: toDoStatus,
```
Comment (zain-sattar) [inline src/components/OneToDo/OneToDo.jsx]: you don't need to send the status on each edit operation just send the text and id and inside 
your editTodo controller function use the spread operator to get the status value

Comment (zain-sattar) [reply src/components/OneToDo/OneToDo.jsx]: similary inside your putToDoInJSON() api function make put request to json server to only update the text of existing toto

File: src/components/OneToDo/OneToDo.jsx [hunk 5]
```diff
@@ -2,30 +2,77 @@ import { Checkbox, IconButton, TextField, Typography } from "@mui/material";
...
+        text: data.ToDoField,
+        checked: toDoList[index].checked,
+        id: toDoList[index].id,
+      };
+      editToDo(editedToDo);
+      editMutation.mutate(editedToDo);
+    }
+  };
+
   const editToDoHandler = () => {
```
Comment (zain-sattar) [inline src/components/OneToDo/OneToDo.jsx]: where are you calling this function?

**Review by zain-sattar — CHANGES_REQUESTED** [review]

File: src/components/OneToDo/OneToDo.jsx (hunk 5 above)
Comment (Abdi-MS) [reply src/components/OneToDo/OneToDo.jsx]: it's to trigger the mutation for reflecting edited changes in server

**Review by zain-sattar — APPROVED** [review]

**Pull request merged**